# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1768756184 -3600
# Node ID 9437c926af55bb109a516f2ceba1c4d40638009c
# Parent  0146250aa5de8f77704fee8f36ac219ea6125fc4
Bug 532643 - Prevent renaming of smart virtual folders. r=darktrojan

This simplifies getting the state of `cmd_renameFolder` by relying on the
`canRename` attribute of the folder and the online state.

Differential Revision: https://phabricator.services.mozilla.com/D276225

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -343,9 +343,8 @@ var folderPaneContextMenu = {
     let isNNTP;
     let isJunk;
     let isVirtual;
     let isInbox;
-    let isSpecialUse;
     let canRenameDeleteJunkMail;
     let isSmartTagsFolder;
     let deletable;
     let server;
@@ -358,9 +357,8 @@ var folderPaneContextMenu = {
     if (multiSelection) {
       canCreateSubfolders = false;
       canRename = false;
       isSmartTagsFolder = false;
-      isSpecialUse = true;
       isInbox = false;
 
       // Set some variables to TRUE to help during the folder lookup loop.
       online = true;
@@ -424,9 +422,8 @@ var folderPaneContextMenu = {
       isNNTP = server.type == "nntp";
       isJunk = flags & Ci.nsMsgFolderFlags.Junk;
       isVirtual = flags & Ci.nsMsgFolderFlags.Virtual;
       isInbox = flags & Ci.nsMsgFolderFlags.Inbox;
-      isSpecialUse = flags & Ci.nsMsgFolderFlags.SpecialUse;
       canRenameDeleteJunkMail = FolderUtils.canRenameDeleteJunkMail(URI);
       isSmartTagsFolder = FolderUtils.isSmartTagsFolder(this.activeFolder);
     }
 
@@ -438,10 +435,9 @@ var folderPaneContextMenu = {
     this._commandStates = {
       cmd_newFolder: online && ((!isNNTP && canCreateSubfolders) || isInbox),
       cmd_deleteFolder:
         online && (isJunk ? canRenameDeleteJunkMail : deletable),
-      cmd_renameFolder:
-        online && ((!isServer && canRename && !isSpecialUse) || isVirtual),
+      cmd_renameFolder: online && canRename,
       cmd_compactFolder:
         !isVirtual && !isNNTP && (isServer || canCompact) && isCompactEnabled,
       cmd_emptyTrash: online && !isNNTP,
       cmd_properties: !multiSelection && !isServer && !isSmartTagsFolder,
